pkg/server/tls/tls.go

```
@@ -0,0 +1,91 @@
+package tls
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"math/big"
+	"net"
+	"time"
+)
+
+// TLSConfig is adapted from http.Server.ServeTLS
+func TLSConfig(hosts []string) (*tls.Config, error) {
+	certPEMBlock, keyPEMBlock, err := generateKeyPair(hosts)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to generate TLS keys %s", err)
+	}
+
+	cert, err := tls.X509KeyPair(certPEMBlock, keyPEMBlock)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to generate X509 key pair %s", err)
+	}
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	}
+
+	return tlsConfig, nil
+}
+
+// Adapted from https://go.dev/src/crypto/tls/generate_cert.go
+func generateKeyPair(hosts []string) ([]byte, []byte, error) {
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		log.Fatal(err)
```

Why not return the error here?

---

Failed to adapt all the code..! Good catch